src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidGetStateCommand.cs

```diff
@@ -53,4 +87,33 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             return Task.FromResult(ExitCode.GENERAL_FAILURE);
         }
     }
+
+    private static StateData GetStateData(ILogger logger)
+    {
+        var runner = new AdbRunner(logger);
+
+        logger.LogDebug("Getting state of ADB and attached Android device(s)");
+
+        var adbVersion = runner.GetAdbVersion()
+            .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+        var state = runner.GetAdbState().Trim();
+
+        var deviceAndArchList = runner.GetAttachedDevicesWithProperties("architecture");

```

Comment:
Suggestion: when parsing some output, it might be useful to provide example output that is to be parsed in comments.

Reply:
Good point, I changed `var` to `Dictionary<string, string?>` so it's easier to read.